app/src/pages/inside/settingsPage/settingsTabs/autoAnalysisTab/analysisForm/strategyBlock/strategyBlock.jsx

```diff
@@ -0,0 +1,114 @@
+import React, { PureComponent } from 'react';
+import classNames from 'classnames/bind';
+import { injectIntl, defineMessages, intlShape } from 'react-intl';
+import { InputBigSwitcher } from 'components/inputs/inputBigSwitcher';
+import { InputRadio } from 'components/inputs/inputRadio';
+import { FormField } from '../formField/formField';
+import styles from './strategyBlock.scss';
+
+const cx = classNames.bind(styles);
+
+const messages = defineMessages({
+  autoAnalysisSwitcherTitle: {
+    id: 'StrategyBlock.autoAnalysisSwitcherTitle',
+    defaultMessage: 'Auto analysis',
+  },
+  strategySelectorTitle: {
+    id: 'StrategyBlock.strategySelectorTitle',
+    defaultMessage: 'Base for Auto Analysis',
+  },
+  analysisOnInfo: {
+    id: 'StrategyBlock.analysisOnInfo',
+    defaultMessage: 'If ON - analysis starts as soon as any launch finished',
+  },
+  analysisOffInfo: {
+    id: 'StrategyBlock.analysisOffInfo',
+    defaultMessage: 'If OFF - not automatic, but can be invoked manually',
+  },
+  sameNameLaunchesCaption: {
+    id: 'StrategyBlock.sameNameLaunchesCaption',
+    defaultMessage: 'Launches with the same name',
+  },
+  sameNameLaunchesInfo: {
+    id: 'StrategyBlock.sameNameLaunchesInfo',
+    defaultMessage:
+      'The test items are analyzed on base of previously investigated data in launches with the same name',
+  },
+  allLaunchesCaption: {
+    id: 'StrategyBlock.allLaunchesCaption',
+    defaultMessage: 'All launches',
+  },
+  allLaunchesInfo: {
+    id: 'StrategyBlock.allLaunchesInfo',
+    defaultMessage:
+      'The test items are analyzed on base of previously investigated data in all launches',
+  },
+});
+
+@injectIntl
+export class StrategyBlock extends PureComponent {
+  static propTypes = {
+    intl: intlShape.isRequired,
+  };
+
+  render() {
+    const { intl } = this.props;
+    return (
+      <React.Fragment>
+        <FormField
+          fieldName="isAutoAnalyzerEnabled"
+          containerClasses={cx('accuracy-form-group')}
+          inputWrapperClasses={cx('switcher-wrapper')}
+          label={intl.formatMessage(messages.autoAnalysisSwitcherTitle)}
+          descriptions={[
```

Comment:
`FormField` is a pure component, but on every render you send a new array, so it doesn't make any sense to be pure (it will rerender every time).
I suggest placing this array (since it doesn't change) in a class field and create it in the constructor.

Reply:
Fixed